pkg/importer/hcl2.go

```diff
@@ -288,6 +292,46 @@ func makeResourceOptions(state *resource.State, names NameTable, addedRefs map[s
 			Value:  cty.True,
 		})
 	}
+	if state.RetainOnDelete {
+		v := cty.BoolVal(state.RetainOnDelete)
+		resourceOptions = appendResourceOption(resourceOptions, "retainOnDelete", &model.LiteralValueExpression{
+			Tokens: syntax.NewLiteralValueTokens(v),
```

Comment:
You know this is `true` because of the `if`, so I think you can just constant-fold this like `Protect` above, as in `NewLiteralValue(cty.True)`, `Value: cty.True`, etc.